### PR TITLE
Apply superadmin privileges to role checks on the server

### PR DESCRIPTION
**Pull request by alxdarksage**
Fix for https://sagebionetworks.jira.com/browse/BRIDGE-2649

### CODE REVIEW

File: src/main/java/org/sagebionetworks/bridge/models/accounts/UserSession.java [hunk 1]
```diff
@@ -101,8 +101,7 @@ public boolean isInRole(Roles role) {
         return (role != null && (proles.contains(SUPERADMIN) || proles.contains(role)));
     }
     public boolean isInRole(Set<Roles> roleSet) {
-        Set<Roles> proles = participant.getRoles();
-        return roleSet != null && (proles.contains(SUPERADMIN) || roleSet.stream().anyMatch(role -> isInRole(role)));
+        return roleSet != null && roleSet.stream().anyMatch(role -> isInRole(role));
```
Comment (DwayneJengSage) [inline src/main/java/org/sagebionetworks/bridge/models/accounts/UserSession.java]: It doesn't look like we have unit tests for SUPERADMIN matching everything in UserSessionTest. Is it worth adding those?

File: src/main/java/org/sagebionetworks/bridge/services/ParticipantService.java [hunk 2]
```diff
@@ -634,14 +629,15 @@ private void updateAccountAndRoles(Study study, Account account, StudyParticipan
             String value = participant.getAttributes().get(attribute);
             account.getAttributes().put(attribute, value);
         }
-        if (callerIsAdmin(callerRoles)) {
-            updateRoles(callerRoles, participant, account);
+        RequestContext requestContext = getRequestContext();
+        if (requestContext.isAdministrator()) {
+            updateRoles(requestContext, participant, account);
```
Comment (DwayneJengSage) [inline src/main/java/org/sagebionetworks/bridge/services/ParticipantService.java:633]: Should we add the superadmin equivalents to adminCanCreateAllRoles() and adminCanUpdateAllRoles() in unit tests?

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]
Overall looks good. Just a few comments about unit tests.

File: src/main/java/org/sagebionetworks/bridge/models/accounts/UserSession.java (hunk 1 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/models/accounts/UserSession.java]: Added

File: src/main/java/org/sagebionetworks/bridge/services/ParticipantService.java (hunk 2 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/services/ParticipantService.java:633]: Added superadminCanCreateEverybody() and superadminCanUpdateEverybody().

**Comment by alxdarksage** [comment]
I also factored out the role tests into a common utility method so future changes are reflected everywhere.

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]
I'm not seeing superadminCanCreateEverybody() and superadminCanUpdateEverybody(). Did we miss a file?

**Comment by alxdarksage** [comment]
Those particular tests were added when I changed the permissions for creating roles on new users (ie. admins can no longer create admins, etc.): https://github.com/Sage-Bionetworks/BridgeServer2/commit/9042501a07420a93f4343d5ea8d4338d0fcddfe9

**Review by DwayneJengSage — APPROVED** [review]

**Pull request merged**